### PR TITLE
unit tests

### PR DESCRIPTION
**Pull request by GaKhalatyan86**

### CODE REVIEW

File: src/app/App.test.js [hunk 1]
```diff
@@ -2,7 +2,5 @@ import { render, screen } from '@testing-library/react';
 import App from './App';
 
 test('renders learn react link', () => {
```
Comment (dmitriisurdin) [inline src/app/App.test.js]: I believe we can delete this test at all

File: src/shared/ui/Counter/Counter.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ export class Counter extends React.Component {
       "div",
       null,
       createElement("button", { className: "btn btn-primary", onClick: this.handleIncrement }, "Increment"),
-      createElement("h1", null, this.state.counter),
+      createElement("h1", {role:"cnt"}, this.state.counter),
```
Comment (dmitriisurdin) [inline src/shared/ui/Counter/Counter.js]: why `cnt` ? :)

File: src/shared/ui/Counter/Counter.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen, fireEvent } from "@testing-library/react";
+import { Counter } from "./Counter";
+
+describe("Counter Component Functionality", () => {
+    test("Renders Initial Value", () => {
```
Comment (dmitriisurdin) [inline src/shared/ui/Counter/Counter.test.js]: the name of the test should stat with 'should' 
e.g. `"Should render initial value"`

File: src/shared/ui/Counter/Counter.test.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        const initialValue = 5;
+        render(<Counter initialValue={initialValue} />);
+        const initialValueElement = screen.getByText(initialValue);
+        expect(initialValueElement).toBeInTheDocument();
+    });
+    test("Decrement Button Functionality", () => {
+        const initialValue = 1;
+        render(<Counter initialValue={initialValue} />);
+        const decrementElement = screen.getByText('Decrement');
+        fireEvent.click(decrementElement);
```
Comment (dmitriisurdin) [inline src/shared/ui/Counter/Counter.test.js]: let's use `userEvent` instead https://blog.mimacom.com/react-testing-library-fireevent-vs-userevent/

File: src/shared/ui/SearchForm/SearchForm.js [hunk 5]
```diff
@@ -19,13 +19,13 @@ export const SearchForm = ({ initialValue = "", onSearch }) => {
   }
 
   return (
-    <form class="search-form input-element">
-      <input type="text" placeholder="Search...."
+    <form className="search-form input-element">
+      <input type="text" role="txtSearch" placeholder="Search...."
```
Comment (dmitriisurdin) [inline src/shared/ui/SearchForm/SearchForm.js]: let's use `data-testid`
e.g `data-testid="search-input"`
`getByTestId("search-input")`

**Review by dmitriisurdin — COMMENTED** [review]
I don't see any cypress tests

File: src/shared/ui/Counter/Counter.js (hunk 2 above)
Comment (GaKhalatyan86) [reply src/shared/ui/Counter/Counter.js]: changed to counter-input

File: src/shared/ui/Counter/Counter.test.js (hunk 3 above)
Comment (GaKhalatyan86) [reply src/shared/ui/Counter/Counter.test.js]: changed

File: src/shared/ui/Counter/Counter.test.js (hunk 4 above)
Comment (GaKhalatyan86) [reply src/shared/ui/Counter/Counter.test.js]: changed

File: src/shared/ui/SearchForm/SearchForm.js (hunk 5 above)
Comment (GaKhalatyan86) [reply src/shared/ui/SearchForm/SearchForm.js]: changed to search-input

File: src/shared/ui/Counter/Counter.test.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import userEvent from '@testing-library/user-event'
+import { act } from 'react-dom/test-utils'
+import { Counter } from "./Counter";
+
+describe("Counter Component Functionality", () => {
+    test("Should render initial value", () => {
+        const initialValue = 5;
+        render(<Counter initialValue={initialValue} />);
+        const initialValueElement = screen.getByText(initialValue);
+        expect(initialValueElement).toBeInTheDocument();
```
Comment (dmitriisurdin) [inline src/shared/ui/Counter/Counter.test.js]: the same result can be achieved with a shorter code
```suggestion
        screen.getByText(mockedInitialValue)
```

File: src/shared/ui/Counter/Counter.test.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen } from "@testing-library/react";
+import userEvent from '@testing-library/user-event'
+import { act } from 'react-dom/test-utils'
+import { Counter } from "./Counter";
+
+describe("Counter Component Functionality", () => {
+    test("Should render initial value", () => {
+        const initialValue = 5;
```
Comment (dmitriisurdin) [inline src/shared/ui/Counter/Counter.test.js]: let's clarify that it's a mock data 
```suggestion
        const mockedInitialValue = 5;
```

considering it's a primitive value, in some projects you can face writing such names using uppercased approach `MOCKED_INITIAL_VALUE`

File: src/shared/ui/SearchForm/SreachForm.test.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { getByTestId, render, screen } from "@testing-library/react";
+import userEvent from '@testing-library/user-event'
+import { SearchForm } from './SearchForm';
+
+describe("SearchForm Component Functionality", () => {
+    test("Should render initial value", () => {
+        const initialValue = "Hello World!";
+        const OnSearchClick = jest.fn();
```
Comment (dmitriisurdin) [inline src/shared/ui/SearchForm/SreachForm.test.js]: in JS we use camel case 
```suggestion
        const onSearchClick = jest.fn();
```

**Review by dmitriisurdin — COMMENTED** [review]
still don't see any cypress tests

File: src/shared/ui/GenreSelect/GenereSelect.test.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen } from "@testing-library/react";
+import userEvent from '@testing-library/user-event'
+import { GenreSelect } from "./GenreSelect";
+import { genreList } from "./GenreMockData";
+
+describe("GenreSelect Component Functionality", () => {
+    test("Should render all genres", () => {
+        const OnGenreClick = jest.fn();
```
Comment (dmitriisurdin) [inline src/shared/ui/GenreSelect/GenereSelect.test.js]: same as above

File: src/shared/ui/GenreSelect/GenereSelect.test.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+                onSearch={OnGenreClick}
+            />
+        );
+        genreList.map((genre) =>
+            expect(screen.getByText(genre)).toBeInTheDocument()
+        );
+    });
+
+    test("Should select genre on button click event", () => {
+        const OnGenreClick = jest.fn((x) => x);
```
Comment (dmitriisurdin) [inline src/shared/ui/GenreSelect/GenereSelect.test.js]: we don't need that for mocked function
```suggestion
        const OnGenreClick = jest.fn();
```

File: src/shared/ui/SearchForm/SearchForm.js [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import React, { useState } from "react"
 import './SearchForm.css'
 
-export const SearchForm = ({ initialValue = "", onSearch }) => {
```
Comment (GaKhalatyan86) [inline src/shared/ui/SearchForm/SearchForm.js]: fix

File: src/shared/ui/Counter/Counter.test.js (hunk 6 above)
Comment (GaKhalatyan86) [reply src/shared/ui/Counter/Counter.test.js]: it didn`t work in my case(

File: src/shared/ui/Counter/Counter.test.js (hunk 7 above)
Comment (GaKhalatyan86) [reply src/shared/ui/Counter/Counter.test.js]: changed to mockedInitialValue

File: src/shared/ui/SearchForm/SreachForm.test.js (hunk 8 above)
Comment (GaKhalatyan86) [reply src/shared/ui/SearchForm/SreachForm.test.js]: changed

File: src/shared/ui/GenreSelect/GenereSelect.test.js (hunk 9 above)
Comment (GaKhalatyan86) [reply src/shared/ui/GenreSelect/GenereSelect.test.js]: changed

File: src/shared/ui/GenreSelect/GenereSelect.test.js (hunk 10 above)
Comment (GaKhalatyan86) [reply src/shared/ui/GenreSelect/GenereSelect.test.js]: removed

File: src/shared/ui/SearchForm/SearchForm.js [hunk 12]
```diff
@@ -12,20 +12,20 @@ export const SearchForm = ({ initialValue = "", onSearch }) => {
...
+  const handleKeyPress = (event) => {
     if (event.key === "Enter") {
       onSearch(searchText);
     }
   }
 
   return (
-    <form class="search-form input-element">
-      <input type="text" placeholder="Search...."
+    <form className="search-form input-element">
```
Comment (dmitriisurdin) [inline src/shared/ui/SearchForm/SearchForm.js]: you don't have any actions for submitting 
```suggestion
    <form className="search-form input-element" onSubmit={()=> onSearch(searchValue)}>
```

File: src/shared/ui/SearchForm/SearchForm.js [hunk 13]
```diff
@@ -12,20 +12,20 @@ export const SearchForm = ({ initialValue = "", onSearch }) => {
...
+      <input type="text" data-testid="search-input" data-cy="search-input" placeholder="Search...."
         value={searchText}
         onChange={handleInputChange}
         onKeyDown={handleKeyPress}
       />
-      <button type="button" onClick={handleSearchClick}>Search</button>
+      <button type="button" data-cy="button-input" onClick={handleSearchClick}>Search</button>
```
Comment (dmitriisurdin) [inline src/shared/ui/SearchForm/SearchForm.js]:
```suggestion
      <button type="submit" data-cy="button-input" >Search</button>
```

https://learn.javascript.ru/forms-submit

File: src/shared/ui/SearchForm/SearchForm.js (hunk 12 above)
Comment (GaKhalatyan86) [reply src/shared/ui/SearchForm/SearchForm.js]: changed

File: src/shared/ui/SearchForm/SearchForm.js (hunk 13 above)
Comment (GaKhalatyan86) [reply src/shared/ui/SearchForm/SearchForm.js]: changed

**Review by dmitriisurdin — APPROVED** [review]

File: src/shared/ui/Counter/Counter.test.js (hunk 6 above)
Comment (dmitriisurdin) [reply src/shared/ui/Counter/Counter.test.js]: take a look at my commit, I fixed it

**Pull request merged**